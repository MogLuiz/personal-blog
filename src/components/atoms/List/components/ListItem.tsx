import { LiHTMLAttributes } from 'react'

import * as S from './styles'

type ListItemProps = LiHTMLAttributes<HTMLLIElement> & {
  children: React.ReactNode
}

export const ListItem = ({ children, ...rest }: ListItemProps) => (
  <S.Container {...rest}>{children}</S.Container>
)
