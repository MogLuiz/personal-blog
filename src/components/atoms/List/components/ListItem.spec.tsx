import { render, screen } from '@testing-library/react'

import { ListItem } from './ListItem'

describe('Components > Atoms - ListItem', () => {
  it('should be able to render the ListItem correctly', () => {
    const { container } = render(<ListItem>item</ListItem>)
    const listChildren = screen.getByText(/item/i)

    expect(listChildren).toBeVisible()
    expect(container.firstChild).toMatchSnapshot()
  })
})
