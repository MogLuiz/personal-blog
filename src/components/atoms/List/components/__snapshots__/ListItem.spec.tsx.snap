// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components > Atoms - ListItem should be able to render the ListItem correctly 1`] = `
<li
  class="text-md relative px-2 font-light transition-colors duration-200 hover:text-link"
>
  item
</li>
`;
